runner/check.go

```diff
@@ -46,6 +51,13 @@ func (c *Check) Run(ctx context.Context, role string) ([]byte, int, error) {
 
 	stdout, stderr, code, err := exec.FullOutput(exec.CommandContext(newCtx, c.Cmd...))
 	if err != nil {
```

Comment:
Could we assume at this point that any error is a check failure, rather than a runner error? If we check our prerequisites that might result in runner errors before we attempt to run the command, then assume that if the command fails it's a check failure rather than an error, we won't need to do potentially brittle things like inspect output for a specific string.

Reply:
well, we could do that, however it would mean we are breaking APIs defined in design doc regarding the *real* errors output.

It just unfortunate that the golang `os./exec` module sends an error on timeout.

Reply:
To @branden point, I've managed to create an ip-detect script that will fail in the condition of a successful return of the contents of the ip-detect script when run, even when the exit code is returned as 22 rather than zero.